lexer/stateful/pattern.go

```diff
@@ -0,0 +1,208 @@
+package stateful
+
+import (
+	"fmt"
+	"regexp/syntax"
+)
+
+///////////////////////////////////////////
```

Comment:
It would be nice to replace this with a more descriptive comment about what the purpose of this is.